pkg/encryption/elgamal/encryption.go

```diff
@@ -0,0 +1,189 @@
+package elgamal
+
+import (
+	crand "crypto/rand"
+	"fmt"
+	"github.com/bwesterb/go-ristretto"
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	"math/big"
+)
+
+type TwistedElGamal struct {
+	curve   *curves.Curve
+	mapping map[string]uint64
+
+	// Notes the sizes decrypted so far. This helps us know if the values we need are already in the map.
+	maxMapping map[MaxBits]bool
+}
+
+func NewTwistedElgamalWithED25519Curve() *TwistedElGamal {
```

Comment:
Can we call this NewDefaultTwistedElgamal instead?

I imagine the users of this library would probably either one of:
us - for building the module in chain - would only use ED25519
- not important since to us since we know what curve our module is using

sei builders - who want to build dapps to interact with the confidential tokens module etc.
- Confusing because they need to figure out what curve to use?

other builders - who just want some encryption library
- not important since they can use whatever curve they want

Reply:
Removed the other constructor, making ED25519 curve the only option, and caching it in the field